add ypbind-mt package

BarrOff opened the pull request:
This is the first time I submit a pull request to nixpkgs.
I read and tried to adhere to the contributing guidelines on a best effort basis.
If there is still something wrong/missing or needs change I will gladly update the PR.

## Description of changes

This pull request adds the package *ypbind-mt* which contains the `ypbind` binary needed for [NIS](https://en.wikipedia.org/wiki/Network_Information_Service) clients to bind to NIS servers.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)

## Misc

The company I work at is using NIS/YP and the necessary tools are still missing in nixpkgs.
This is intended to be the first in a series of packages needed to create usable NIS clients (yp-tools and libnss_nis will follow).
This deviation was created and successfully built/used on a NixOS machine.

A few changes to the Makefile for manpages are necessary.
 I did not find a way to prevent xmllint from trying to access the network.
Maybe there is a better way than commenting out the respective lines using `sed`?

Best regards
BarrOff

BarrOff committed:
maintainers: add BarrOff

AndersonTorres commented on maintainers/maintainer-list.nix:
```diff
@@ -2249,6 +2249,11 @@
+    name = "BarrOff";
+    github = "BarrOff";
+    githubId = 58253563;
+  };
```
e-mail, please

BarrOff replied:
I would prefer not to add my private mail to a public repo.
I checked the maintainer list and there are other entries consisting only of name, github, githubId so I thought this would be possible?.

AndersonTorres replied:
Usually I recommend a specialized e-mail for this.
But let's not pursue this now.

AndersonTorres commented on pkgs/by-name/yp/ypbind-mt/package.nix:
```diff
@@ -0,0 +1,74 @@
+    hash = "sha256-P8LH1vBTrNdgUQhzeixxBZR4atXyRVF+DuUbR6UeBJY=";
+  };
+
+  outputs = [ "out" ];
```
```suggestion
```

BarrOff replied:
Has been fixed in my refactor commit [`97c8bec` (#333099)](https://github.com/NixOS/nixpkgs/pull/333099/commits/97c8bec445635a1a843d91f7d1eac67e274f9565)

AndersonTorres commented on pkgs/by-name/yp/ypbind-mt/package.nix:
```diff
@@ -0,0 +1,74 @@
+    systemdLibs
+  ];
+
+  src = fetchFromGitHub {
```
usually src comes after version

BarrOff replied:
Has been fixed in [`8e49884` (#333099)](https://github.com/NixOS/nixpkgs/pull/333099/commits/8e498848af210a5a743e44fbfd5ac4f28d657969)

AndersonTorres commented on pkgs/by-name/yp/ypbind-mt/package.nix:
```diff
@@ -0,0 +1,74 @@
+
+  outputs = [ "out" ];
+
+  configurePhase = ''
```
Can you explain your code here?

BarrOff replied:
Without the `configurePhase/installPhase` settings the derivation tried to install the files to the global root `/` directory.
 Has been fixed in my refactor commit [`97c8bec` (#333099)](https://github.com/NixOS/nixpkgs/pull/333099/commits/97c8bec445635a1a843d91f7d1eac67e274f9565)

AndersonTorres commented:
> I did not find a way to prevent xmllint from trying to access the network.

Roughly speaking, XMLLINT requires some files (in a XSLT format) so that it knows what to do - think on it as providing a map with instructions to a human person.
If those files/maps are not locally present, then xmllint tries to download them (like a person opening the Maps applet).

Then you need to provide these XSLT files somehow. Usually Nixpkgs provides them already.

BarrOff commented:
The Makefile has a hardcoded http url to a xml style sheet, see [here](https://github.com/thkukuk/ypbind-mt/blob/master/man/Makefile.am) and I see two solutions:

1. Disable this by commenting out that code (my initial solution)
2. Replace the http url with the `manpage/docbook.xsl` file from the `docbook-xsl-nons` package

By chance I tried the official release tarball from the repos release page and it worked without the problems I had with the `fetchFromGithub` code.
There the `configure` script automatically comments out the problematic lines.
It is also able to build the code without any custom configPhase/installationPhase settings.

I think this is the preferable solution and made a commit for this.

@AndersonTorres Do you see any remaining issues with the PR?

AndersonTorres commented:
> I think this is the preferable solution and made a commit for this.

Usually we prefer to use the pristine source code, so that we have more control over the build process.
It became more pronounced since the [xzdoor incident](https://github.com/Getshell/xzDoor).

But for now let's pick the tarball.

AndersonTorres commented on pkgs/by-name/yp/ypbind-mt/package.nix:
```diff
@@ -0,0 +1,45 @@
+    systemdLibs
+  ];
+
+  meta = with lib; {
+    description = "Multithreaded daemon maintaining the NIS binding informations.";
+    homepage = "https://github.com/thkukuk/ypbind-mt";
+    changelog = "https://github.com/thkukuk/ypbind-mt/blob/master/NEWS";
+    license = licenses.gpl2;
+    platforms = platforms.linux;
+    mainProgram = "ypbind";
+    maintainers = with maintainers; [ BarrOff ];
+  };
```
```suggestion
  meta = {
    description = "Multithreaded daemon maintaining the NIS binding informations.";
    homepage = "https://github.com/thkukuk/ypbind-mt";
    changelog = "https://github.com/thkukuk/ypbind-mt/blob/master/NEWS";
    license = lib.licenses.gpl2Plus;
    mainProgram = "ypbind";
    maintainers = with lib.maintainers; [ BarrOff ];
    platforms = lib.platforms.linux;
  };
```

AndersonTorres reviewed (changes requested):
Besides, LGTM

AndersonTorres commented:
Now get rid of that excess of Semantic Commits and follow our commit conventions:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

BarrOff commented:
Just to make sure I understand correctly how to proceed:

1. I should do a rebase of all my commits to adjust the commit messages, then force push to my branch?
2. Am I supposed to squash some commits where suitable?

AndersonTorres commented:
Yes. Get rid of that truckload of "feat/fix/style/etc".

Only things are
- maintainers: add <handle>
- <pkg>: init at <version>

BarrOff committed:
ypbind-mt: init at 2.7.2

BarrOff commented:
Most projects I have worked on before use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) so I used that.
In the meantime I read the contributing guide again and noticed the commit style guide.
I have rebased and squashed the commits as requested.

AndersonTorres reviewed (approved):
Result of `nixpkgs-review pr 333099` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ypbind-mt</li>
  </ul>
</details>
